test/simd/test_simd_fnt.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2017-2018 Scality
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <vector>
+
+#include <functional>
+#include <gtest/gtest.h>
+
+#include "arith.h"
+#include "core.h"
+#include "misc.h"
+
+#ifdef QUADIRON_USE_SIMD
+
+#include "simd.h"
+#include "simd/simd.h"
+#include "simd_fnt.h"
+
+namespace simd = quadiron::simd;
+
+template <typename T>
+class SimdTestFnt : public ::testing::Test {
+  public:
+    SimdTestFnt()
+    {
+        if (sizeof(T) == 2) {
+            this->q = 257;
+        } else if (sizeof(T) == 4) {
+            this->q = static_cast<T>(65537);
+        } else {
+            throw "Wrong TypeParam for SimdTestFnt tests";
+        }
+
+        this->distribution =
+            std::make_unique<std::uniform_int_distribution<uint32_t>>(0, q - 1);
+    }
+
+    simd::VecType rand_vec(T lower = 0, T upper_bound = 0)
+    {
+        const size_t n = simd::countof<T>();
+        T buf[n];
+        simd::VecType* vec = reinterpret_cast<simd::VecType*>(buf);
+
+        T bound = upper_bound ? upper_bound : q;
+        bound -= lower;
+
+        for (unsigned i = 0; i < n; i++) {
+            buf[i] = lower + distribution->operator()(quadiron::prng()) % bound;
```

Comment:
This is a biaised way to generate a random number in a range.

It would be better to create a new `uniform_int_distribution`, no?

Reply:
it's not necessary but we do better like that.